test/test-app-folder-artifact-same.js

```diff
@@ -0,0 +1,150 @@
+        .withPrompts({
+          sdkVersion: '2.1.1',
+          projectArtifactId: 'test-artifact',
+        })
+        .on('end', done);
+    }.bind(this));
+
+    it('creates files', function () {
+      assert.file([
+        '.yo-rc.json',
+      ]);
+    });
+    it('parent folder should equal to artifactId', function() {
+      assert.equal(path.basename(process.cwd()),'test-artifact');
+    });
+    it('grand-parent folder should be temp-test', function() {
+      assert.equal(path.basename(path.resolve('..')),'temp-test');
+    });
+    it('re-running the install with warning=true', function(done) {
+      helpers.run(path.join(__dirname, '../generators/app'))
+        // generator will create a temp directory and make sure it's empty
+        .inTmpDir(function() {
+          // we want our test to run inside the previously generated directory
+          // and we don't want it to be empty, so this is a hack for that.
+          process.chdir(path.join(this.osTempDir, 'test-artifact'));
+        }.bind(this))
+        .withPrompts({
+          warning: true,
+          sdkVersion: '2.2.0',
+          projectArtifactId: 'test-artifact-1',
+        })
+        .on('end', function() {
+          assert.equal(fs.existsSync(path.join(this.osTempDir,'test-artifact/test-artifact-1')),false);
+          done();
+        }.bind(this));
+    }.bind(this));
+    it('re-running the install with warning=false and abortProjectArtifactIdUpdate=true', function(done) {
+      helpers.run(path.join(__dirname, '../generators/app'))
+        // generator will create a temp directory and make sure it's empty
+        .inTmpDir(function() {
+          // we want our test to run inside the previously generated directory
+          // and we don't want it to be empty, so this is a hack for that.
+          process.chdir(path.join(this.osTempDir, 'test-artifact'));
+        }.bind(this))
+        .withPrompts({
+          warning: false,
+          sdkVersion: '2.2.0',
+          projectArtifactId: 'test-artifact-1',
+          abortProjectArtifactIdUpdate: true,
+        })
+        .on('end', function() {
+          assert.equal(fs.existsSync(path.join(this.osTempDir,'test-artifact/test-artifact-1')),false);
+          done();
+        }.bind(this));
+    }.bind(this));
+  });
+
+  describe('artifactId == folder name for new project', function () {
+
+    this.timeout(60000);
+    this.osTempDir = path.join(os.tmpdir(), 'demo');
+
+    before(function (done) {
+      helpers.run(path.join(__dirname, '../generators/app'))
+        .inDir(this.osTempDir)
+        .withOptions({ 'skip-install': true })
+        .withPrompts({
+          sdkVersion: '2.1.1',
+          projectArtifactId: 'demo',
+        })
+        .on('end', done);
+    }.bind(this));
+    it('creates files', function () {
+      assert.file([
+        '.yo-rc.json',
+        this.osTempDir,
+      ]);
+    }.bind(this));
+    it('sub-folder is not created', function () {
```

Comment:
Rename this test? 'no artifactId sub-folder is created because the current folder is named the same as the artifactId'